idl/github.com/uber/cadence/shared.thrift

```diff
@@ -871,6 +871,15 @@ struct RecordActivityTaskHeartbeatRequest {
   30: optional string identity
 }
 
+struct RecordActivityTaskHeartbeatByIDRequest {
+  10: optional string domain
+  20: optional string workflowID
```

Comment:
do we conventionally use gen.WorkflowExecution?

Reply:
you are right, I will follow other XXByID for now, and change everywhere not use WorkflowExecution later